python/ray/util/spark/cluster_init.py

```diff
@@ -1395,6 +1446,37 @@ def shutdown_ray_cluster() -> None:
         _active_ray_cluster = None
 
 
+@PublicAPI(stability="alpha")
+def serve_global_ray_cluster(
```

Comment:
When is this API called?

Reply:
User need to create a separate databricks notebook to run the API, then the Ray global cluster is up and the notebook must keep attached and running , when the notebook is detached, the Ray global cluster is down

Reply:
I feel like this is a complex user experience for them, and it's not that stable. Could we send a spark job for running this on cluster configuration page when starting the cluster?

Reply:
> Could we send a spark job for running this on cluster configuration page when starting the cluster?

to run a spark job, we need to create a notebook on the cluster firstly, and then run the notebook, otherwise we cannot get the spark session to create spark job

another way is config a job on databricks cluster https://docs.databricks.com/en/workflows/jobs/create-run-jobs.html

but I think these are follow-up tasks, we should add the API `serve_global_ray_cluster` firstly

Reply:
> Will this work if a user calls an ephemeral notebook from within their interactive notebook to start the ray cluster?

No , this is blocking API, once it returns, the Ray cluster is shut down


> can we ensure that this API can only be started from an interactive notebook that is accessible and capable of being interacted with by a user?

Do we have databricks API to check whether it is interactive notebook ? If yes we can do it.

Reply:
@BenWilson2 

Tested, calling an ephemeral notebook to start the ray cluster works: See

https://e2-dogfood.staging.cloud.databricks.com/?o=6051921418418893#notebook/1069431269500840/command/1069431269500850

Reply:
So in order to run global ray cluster, someone needs to run it in one notebook and make sure the notebook keeps running? Isn't that too fragile. Shouldn't we decouple the global ray cluster lifecycle with the notebook lifecycle so it's more like a daemon.

Reply:
> So in order to run global ray cluster, someone needs to run it in one notebook and make sure the notebook keeps running? Isn't that too fragile. Shouldn't we decouple the global ray cluster lifecycle with the notebook lifecycle so it's more like a daemon.

We can also make it run in a databricks job. Currently only in databricks notebook or databricks job we can get the "spark session" (only using the spark session, we can request databricks cluster resources) . 

I think this is databricks side work we need to improve, but in Ray code side, we need to add a "serve_ray_global_cluster" API anyway.